src/app/pages/subjects/new/subjects-listing.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import useSubjectsContext, {SubjectData} from './context';
+import useSubjectCategoryContext from '~/contexts/subject-category';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {faArrowRight} from '@fortawesome/free-solid-svg-icons/faArrowRight';
+import {useLocation} from 'react-router-dom';
+import './subjects-listing.scss';
+
+export default function SubjectsListing() {
+    const {subjects} = useSubjectsContext();
+    const names = Reflect.ownKeys(subjects) as Array<string>;
```

Comment:
This is interesting, I actually haven't seen Reflect before, I'm guessing this is better than Object.keys?

Reply:
Arguably better [in some ways](https://blog.greenroots.info/javascript-why-reflect-apis). Reflect is the ES6 package that collects all the introspection stuff as static functions rather than Class methods (there's no instantiable Reflect object).

Reply:
Thanks for the explanation!